internal/run/wrapper.go

```diff
@@ -47,6 +50,41 @@ func (a *API) ReplaceService(namespace, serviceID string, svc *run.Service) (*ru
 	return a.Client.Namespaces.Services.ReplaceService(serviceName, svc).Do()
 }
 
+// ServicesWithLabelSelector gets services filtered by a label selector.
+func (a *API) ServicesWithLabelSelector(namespace string, labelSelector string) ([]*run.Service, error) {
+	parent := fmt.Sprintf("namespaces/%s", namespace)
+
+	servicesList, err := a.Client.Namespaces.Services.List(parent).LabelSelector(labelSelector).Do()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to filter services by label selector")
+	}
+
+	return servicesList.Items, nil
```

Comment:
this is something to learn, you can do:

`return servicesList.Items, errors.Wrap(err, "..")`

Wrap returns nil if err is nil. But make sure `servicesList` isn't a pointer type that can panic in case of error.

Reply:
That's helpful and I didn't know about it. I wonder if in this case it is better as it is since `servicesList` is a pointer, so I will have add an if anyways. Since is more common to see check for errors, I think an `if err != nil` make it more readable.

Reply:
If servicesList.Items has a chance of panicking, don't do what I said. :) it works otherwise however